### PR TITLE
General Heretics Bugfixing Patch

### PR DESCRIPTION
**Pull request by Djiq**
## About The Pull Request

This pr fixes a few things if i remember correctly.

- Slightly reworks armsy to use get_turf() instead of .loc to avoid it's chain-link fuckery.
- Changes how reality_smash_tracker works so it reapplies huds to new heretics, so they can actually see the influences.
- Fixes a runtime caused by heretics getting gibbed.
- Fixes the transmutation rune breaking every time you made a voiceless dead.

this pr also slightly reduces Void Grasp's silence duration, the change was so small it didnt warrant it's own pr.

## Why It's Good For The Game

Fixes gud

## Changelog
:cl:
fix: Fixed armsy breaking sometimes.
fix: Fixed heretics not seeing some influences sometimes.
fix: Heretics will no longer runtime like crazy when they die.
fix: Voiceless dead should no longer break the voiceless ritual.
balance: slightly lowers void silent duration.
/:cl:

### CODE REVIEW

File: code/modules/antagonists/eldritch_cult/eldritch_effects.dm [hunk 1]
```diff
@@ -7,7 +7,6 @@
 	layer = SIGIL_LAYER
 	///Used mainly for summoning ritual to prevent spamming the rune to create millions of monsters.
 	var/is_in_use = FALSE
-
 /obj/effect/eldritch/Initialize()
```
Comment (Timberpoes) [inline code/modules/antagonists/eldritch_cult/eldritch_effects.dm:10]: No newline uwu?

**Review by Timberpoes — CHANGES_REQUESTED** [review]

File: code/modules/antagonists/eldritch_cult/eldritch_effects.dm (hunk 1 above)
Comment (EdgeLordExe) [reply code/modules/antagonists/eldritch_cult/eldritch_effects.dm:10]: uwah

File: code/modules/antagonists/eldritch_cult/eldritch_effects.dm [hunk 2]
```diff
@@ -222,20 +226,21 @@
 	anchored = TRUE
 	resistance_flags = FIRE_PROOF | UNACIDABLE | ACID_PROOF
 	invisibility = INVISIBILITY_OBSERVER
+	var/datum/atom_hud/alternate_appearance/basic/heretics/alt_appearance
```
Comment (AnturK) [inline code/modules/antagonists/eldritch_cult/eldritch_effects.dm:229]: Needs cleanup for clients and itself in destroy.

**Review by AnturK — CHANGES_REQUESTED** [review]

File: code/__DEFINES/antagonists.dm [hunk 3]
```diff
@@ -72,8 +72,8 @@
 #define CONTRACT_UPLINK_PAGE_HUB "HUB"
 
 ///It is faster as a macro than a proc
-#define IS_HERETIC(mob) (mob.mind?.has_antag_datum(/datum/antagonist/heretic))
-#define IS_HERETIC_MONSTER(mob) (mob.mind?.has_antag_datum(/datum/antagonist/heretic_monster))
+#define IS_HERETIC(mob) (mob?.mind?.has_antag_datum(/datum/antagonist/heretic))
```
Comment (Rohesie) [inline code/__DEFINES/antagonists.dm:75]: This check shouldn't be in the define.

File: code/modules/antagonists/eldritch_cult/eldritch_effects.dm (hunk 1 above)
Comment (Rohesie) [reply code/modules/antagonists/eldritch_cult/eldritch_effects.dm:10]: No need to save on lines. It's better if there's some sort of separation, so it's clear where one definition ends and when another starts.

File: code/modules/antagonists/eldritch_cult/eldritch_antag.dm [hunk 4]
```diff
@@ -87,7 +88,7 @@
 
 /datum/antagonist/heretic/process()
 
-	if(owner.current.stat == DEAD)
+	if(owner?.current?.stat == DEAD)
```
Comment (Rohesie) [inline code/modules/antagonists/eldritch_cult/eldritch_antag.dm:91]: Why is this check necessary?*

**Pull request closed without merging**